### DISCUSSION TITLE
Request for the Astro ProgressCheck, Quiz & Checklist component

### DISCUSSION THREAD

**Discussion by SnowDingo** [withastro/starlight#2151] [Feature Requests]
### What version of `starlight` are you using?

The latest

### What is your idea?
I have already done research about this in the discussion "New components for Astro", but I believe that there should be all three of them so I created this discussion.
As used in the official Astro Document, Starlight should add three new components as used in the official Astro.js document: The Progress check component, the checklist component, and the test your knowledge/quiz component.

<img width="235" alt="スクリーンショット 2024-07-25 9 14 16" src="https://github.com/user-attachments/assets/b37ecc25-2652-4bcc-8397-64ba4f37338b">

<img width="641" alt="スクリーンショット 2024-07-25 9 14 48" src="https://github.com/user-attachments/assets/0445569b-f7ff-4e6b-a049-c098fc07cbf7">

<img width="654" alt="スクリーンショット 2024-07-25 9 25 08" src="https://github.com/user-attachments/assets/3b11df62-803b-42b2-8ec4-5eeff9a3529d">

### Why is this feature necessary?

This feature is necessary because it provides a new possibility for Starlight documentation.
It allows documents to have better tutorials and provides interactivity to the docs. In addition, it is easy to add the component since it is already fully coded in the Astro official document.

### Do you have examples of this feature in other projects?

- The official Astro Doc's "Build a blog tutorial"

### Participation

- [X] I am willing to submit a pull request for this feature.

**Comment by rajeevmyname** [comment]
I am looking for the similar components as well

**Comment by Barishhh** [comment]
Any status on this?

**Comment by azeveco** [comment]
Hey guys!

While Starlight does not include these components, the [starlight-videos](https://github.com/HiDeoo/starlight-videos) plugin adds them.

You can check more about them:
- [Checklist/List](https://starlight-videos.netlify.app/components/list/)
- [Quiz](https://starlight-videos.netlify.app/components/quizz/)

Now talking about this progress component, you could extract it from the official docs to a plugin and share it with the community. Since you're willing to build it and this is an old discussion, I don't why not.
